Add Vial support to ProjectD Tinker 65 ANSI

FlamingPhoenix40 opened the pull request:
Added Vial support to the ProjectD Tinker65, as the entry in the keyboards folder seems to be the version from the normal QMK repo. Had to reduce the amount of layers from 13 to 4 to make it fit in the EEPROM. Can possibly be made to work with more layers, but haven't tested with more than 4 because I don't need more than that. Initially disabled some features to try and make fit, and for personal use kept it that way to squeeze out some extra free space for macros, but decided to leave enabled by default for this PR. Can be disabled again by users by uncommenting lines in rules.mk.

This is my first PR ever, so please do tell me if I did something wrong :)

FlamingPhoenix40 committed:
Add support for ProjectD Tinker65 ANSI

Reduced nr. of layers from 13 to 4 and disabled some features to save space, because otherwise wouldn't fit EPROM

FlamingPhoenix40 committed:
All features enabled by default

All features now enabled by default, and only nr. of layers reduced to make fit. Can uncomment lines in rules.mk to squeeze a bit more free space out if features are not needed.

xyzz commented:
fails to build

FlamingPhoenix40 committed:
Fix nr. of layers in keyboard.json

When uploading to github for the pull request, forgot to edit the keyboard.json to decrease the number of layers. Only did that in my local offline clone.

FlamingPhoenix40 commented:
Apologies. When testing on a locally downloaded copy of the repository, I made a change to the keyboard.json to reduce the number of layers, because using `#define DYNAMIC_KEYMAP_LAYER_COUNT` in the keymaps config.h didn't work, it got overridden, and I don't have enough knowledge to fix it other than editing the nr of layers in the keyboard.json. When uploading to my own fork for this PR, I forgot to upload the edited keyboard.json and only provided the new keymap. 

I pushed the corrected version to my fork, so it should work again now.

xyzz commented on keyboards/projectd/65/projectd_65_ansi/keyboard.json:
```diff
@@ -8,7 +8,7 @@
     "diode_direction": "COL2ROW",
     "dynamic_keymap": {
-        "layer_count": 13
+        "layer_count": 4
```
this would not typically be acceptable since it will conflict on qmk merges. have you tried to override it in the individual keymap by using json?

FlamingPhoenix40 replied:
I've tried reducing the keymap number by putting `#define DYNAMIC_KEYMAP_LAYER_COUNT 4` in the keymaps/vial/config.h file, as per the instructions [on the vial website](https://get.vial.today/docs/firmware-size.html) about reducing firmware size, but when I do it that way, the map fails to build with an error about the keymap count being redefined:

```
 <command-line>:
./keyboards/projectd/65/projectd_65_ansi/keymaps/vial/config.h:10: error: "DYNAMIC_KEYMAP_LAYER_COUNT" redefined [-Werror]
   10 | #define DYNAMIC_KEYMAP_LAYER_COUNT 4
      |
In file included from <command-line>:
./.build/obj_projectd_65_projectd_65_ansi_vial/src/info_config.h:37: note: this is the location of the previous definition
   37 | #    define DYNAMIC_KEYMAP_LAYER_COUNT 13
      |
cc1.exe: all warnings being treated as errors
 [ERRORS]
 |
 |
 |
make[1]: *** [builddefs/common_rules.mk:365: .build/obj_projectd_65_projectd_65_ansi_vial/.build/obj_projectd_65_projectd_65_ansi_vial/src/community_modules.o] Error 1
Make finished with errors
```

That's why I edited the layer count in the keyboard.json. I've tried changing it in the keymap json by adding the code snippet below, but then it fails to build because it doesn't fit in EEPROM when it wants to compile quantum/dynamic_keymap.c

```
 "dynamic_keymap": {
      "layer_count": 4
    },
```

I'm not experienced enough with QMK or Vial to how else to fix it. If I am not able to fix it another way, and the current way I did it is not acceptable, I'll drop the PR and wait for someone else to fix it. I was happy to get it to work on my own board and thought I'd contribute haha

xyzz replied:
try to add `#undef DYNAMIC_KEYMAP_LAYER_COUNT` before the `#define` maybe it would work

FlamingPhoenix40 replied:
Jep, that seemed to fix the issue, pushing now. Thank you for the assist!

FlamingPhoenix40 committed:
Moved layer reduction to vial/config.h and reverted main keyboard.json

Reverted main keyboard.json back to default for upstream QMK compatibility. Put redefinition of layers back in vial/config.h but added a #undef to get rid of the definition from keyboard.json that caused this method to not build.

shyguy80 commented:
@FlamingPhoenix40 How difficult would it be to port this to the ISO (Qwertz) Version of this Keyboard? Can you give me some starting Hints?
Edit: I uploaded a modified version in a new issue. Maybe you could check the files..
